src/main/java/io/localmotion/picture/sanitizer/PictureSanitizerImplementation.java

```diff
@@ -0,0 +1,109 @@
+package io.localmotion.picture.sanitizer;
+
+import io.localmotion.picture.compressor.PictureCompressor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.imaging.ImageInfo;
+import org.apache.commons.imaging.Imaging;
+
+import javax.imageio.ImageIO;
+import javax.imageio.ImageReader;
+import javax.imageio.stream.ImageInputStream;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+
+@Singleton
+@Slf4j
+public class PictureSanitizerImplementation implements PictureSanitizer {
+
+
+	@Inject
+	PictureCompressor pictureCompressor;
+
+	@Override
+	public boolean trySanitize(File file) {
+		boolean safeState = false;
+		boolean fallbackOnApacheCommonsImaging;
+		try {
+			String formatName = null;
+			if((file != null) && file.exists() && file.canRead() && file.canWrite()) {
+				try(ImageInputStream imageInputStream = ImageIO.createImageInputStream(file)) {
+					Iterator<ImageReader> imageReaderIterator = ImageIO.getImageReaders(imageInputStream);
+					if ( imageReaderIterator.hasNext()) {
+						ImageReader imageReader = imageReaderIterator.next();
```

Comment:
Why just process the first iteration? What would come after that, that we are skipping now?